lib/Doctrine/Persistence/Mapping/MappingException.php

```diff
@@ -92,4 +92,14 @@ public static function nonExistingClass($className)
     {
         return new self(sprintf("Class '%s' does not exist", $className));
     }
+
+    /**
+     * @param string $className
+     *
+     * @return MappingException
+     */
+    public static function anonymousClass($className)
```

Comment:
We can use type declarations here:

```suggestion
    public static function anonymousClass(string $className): self
```

Reply:
Added type hinting. But, this class in other methods don't use type hinting. As result, I don't add previously.

Reply:
That's because they were introduced at a time when we could not use them. We are probably going to migrate those at some point, so let's do it rightaway for new methods, it will mean less work for the migration.